sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp

```diff
@@ -160,24 +160,65 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     /**
      * @brief Sets POSIX access control rights on files and directories under given directory
      * recursively.
-     * @param mode Mode PathSetAccessControlRecursiveMode::Set sets POSIX access control rights on
-     * files and directories, PathSetAccessControlRecursiveMode::Modify modifies one or more POSIX
-     * access control rights  that pre-exist on files and directories,
-     * PathSetAccessControlRecursiveMode::Remove removes one or more POSIX access control rights
-     * that were present earlier on files and directories
      * @param acls Sets POSIX access control rights on files and directories. Each access control
      * entry (ACE) consists of a scope, a type, a user or group identifier, and permissions.
      * @param options Optional parameters to set an access control recursively to the resource the
      * directory points to.
-     * @return Azure::Core::Response<Models::SetDataLakeDirectoryAccessControlRecursiveResult>
+     * @return
+     * Azure::Core::Response<Models::SetDataLakeDirectoryAccessControlRecursiveSinglePageResult>
      * @remark This request is sent to dfs endpoint.
      */
-    Azure::Core::Response<Models::SetDataLakeDirectoryAccessControlRecursiveResult>
-    SetAccessControlRecursive(
-        Models::PathSetAccessControlRecursiveMode mode,
+    Azure::Core::Response<Models::SetDataLakeDirectoryAccessControlRecursiveSinglePageResult>
+    SetAccessControlRecursiveSinglePage(
         std::vector<Models::Acl> acls,
-        const SetDataLakeDirectoryAccessControlRecursiveOptions& options
-        = SetDataLakeDirectoryAccessControlRecursiveOptions()) const;
+        const SetDataLakeDirectoryAccessControlRecursiveSinglePageOptions& options
+        = SetDataLakeDirectoryAccessControlRecursiveSinglePageOptions()) const
+    {
+      return SetAccessControlRecursiveSinglePageInternal(
+          Models::PathSetAccessControlRecursiveMode::Set, acls, options);
+    }
+
+    /**
+     * @brief Updates POSIX access control rights on files and directories under given directory
+     * recursively.
+     * @param acls Updates POSIX access control rights on files and directories. Each access control
+     * entry (ACE) consists of a scope, a type, a user or group identifier, and permissions.
+     * @param options Optional parameters to set an access control recursively to the resource the
+     * directory points to.
+     * @return
+     * Azure::Core::Response<Models::UpdateDataLakeDirectoryAccessControlRecursiveSinglePageResult>
+     * @remark This request is sent to dfs endpoint.
+     */
+    Azure::Core::Response<Models::UpdateDataLakeDirectoryAccessControlRecursiveSinglePageResult>
+    UpdateAccessControlRecursiveSinglePage(
```

Comment:
what's the difference between set and update?

Reply:
I don't know, it is server behavior.

Reply:
so are you gonna write unit test?

Reply:
I don't understand, as developer of this SDK, we don't even know the behavior of the API we wrote?

Reply:
Sure, I was trying to do it fast due to the deadline we are catching, but I agree that not understanding the API clearly is not acceptable, I'll reset my attitude.

Reply:
`Update` will only be setting the matching type/ID of the ACL, while `Set` will overwrite all ACLs.
`Delete`, as I previously guessed but not confirmed, is confirmed to be only deleting the specified ACL.